src/libsyntax/attr.rs

```diff
@@ -356,25 +358,79 @@ pub fn requests_inline(attrs: &[Attribute]) -> bool {
     }
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct GatedCfg {
+    span: Span,
+    kind: GatedCfgKind,
+}
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+enum GatedCfgKind {
+    TargetFeature
+}
```

Comment:
Hm, well either that or perhaps something along the lines of an [iterator macro](https://github.com/rust-lang/cargo/blob/master/src/bin/cargo.rs#L63-L88) where each instance you need to match below just [provides its own macro](https://github.com/rust-lang/cargo/blob/master/src/bin/cargo.rs#L142-L154) for generating code.